### DISCUSSION TITLE
Create Legend in SKCartesianChart

### DISCUSSION THREAD

**Discussion by essenius** [Live-Charts/LiveCharts2#123] [Q&A] [answered]
I'm creating a function to dynamically create a cartesian chart that and serve it as an image. I'm using SKCartesianChart for that, and it works pretty nicely. What I have not been able to figure out is how to create a legend for this chart. Since the application I write is headless, I don't have UI components to tie into. I checked the examples, but as far as I could see they all seem to assume there is a UI. How do I create a legend in only code?

**Comment by beto-rodriguez** [comment, marked as answer]
`SKCartesianChart`, `SKPieChart` and `SKGeoMap` classes do not require a UI.

But for now all the legends and tooltips are user controls on each platform, this allows the user to pass a custom template to render their own tooltips and legends using the syntax in their current platform (Xamarin, Avalonia, WPF, WinUI ...) sadly there is no a standard between all these platforms.

An alternative is to create your own `IChartLegend<T>` and use the SkiaSharp API to draw the legend, this way we could share the legend across all platforms (and "in-memory charts" also such as the `SKCharts` classes). Actually I was going for this approach for the library, but then it would have been harder for the user to customize legends and tooltips.

**Reply by beto-rodriguez** [reply]
We can track the implementation of the default SkiaSharp based legend at #144.
